### PR TITLE
Safer type modification

### PR DESCRIPTION
**Pull request by GeorgeR227**
This should help solve problems in type inference regarding typing rules doing strange things. Addresses problems but does not fully resolve issue #46.

### CODE REVIEW

**Comment by GeorgeR227** [comment]
To be clear, this should resolve the issues present in that issue but a MWE and proper tests are still not available.

**Comment by lukem12345** [comment]
I would change it such that `safe_modifytype` is mutating, and just returns true or false, unless its too unseemly to pass the necessary ref. 

Adding the Halfar equation with `avg_01` ought to suffice for now, and the code cov report is an automatic check to confirm that this is getting hit.

**Comment by GeorgeR227** [comment]
I decided to make a modifying wrapper around the function. Reason being that I want the core of the function itself to be very short and concise, plus leaving it like this makes testing it easier since I don't need to build up Decapodes for it.

File: src/acset.jl [hunk 1]
```diff
@@ -341,7 +341,18 @@ function add_parameter(d::AbstractNamedDecapode, k::Symbol)
     return add_part!(d, :Var, type=:Parameter, name=k)
 end
 
+function safe_modifytype(tgt_type::Symbol, src_type::Symbol)
```
Comment (lukem12345) [inline src/acset.jl:344]: A better name should be picked instead of using "src" and "tgt". Adding a comment above this line will help clarify as well.

Comment (GeorgeR227) [reply src/acset.jl:344]: I've changed this to be `org_type` and `new_type` respectively, plus added docstrings. I've looked at too much assembly before I guess.

Comment (lukem12345) [reply src/acset.jl]: The problem is src and tgt are just too good and then get used everywhere

**Comment by lukem12345** [comment]
Want to merge this now and do a separate PR for Constant?

**Comment by GeorgeR227** [comment]
Yeah that's the plan. The Constant behavior is baked into the type inference so changing that seems more involved.

**Review by lukem12345 — APPROVED** [review]

**Comment by GeorgeR227** [comment]
Close #46. We now have a MWE and a test to cover it.

**Pull request merged**